src/Cedar/Proto_PPP.c

```diff
@@ -1339,29 +1346,43 @@ bool PPPProcessEAPResponsePacket(PPP_SESSION *p, PPP_PACKET *pp, PPP_PACKET *req
 			{
 			case AUTHTYPE_RADIUS:
 				// Create EAP client if needed
-				if (p->MsChapV2_UseDoubleMsChapV2 && p->EapClient == NULL)
+				if (p->EapClient == NULL)
 				{
 					char client_ip_tmp[256];
+					PPP_LCP *response = NULL;
 					IPToStr(client_ip_tmp, sizeof(client_ip_tmp), &p->ClientIP);
-					Debug("EAP-MSCHAPv2 creating EAP RADIUS client\n");
-					p->EapClient = HubNewEapClient(p->Cedar, p->Eap_Identity.HubName, client_ip_tmp, p->Eap_Identity.UserName, "L3:PPP");
+					Debug("Creating EAP RADIUS client\n");
+					p->EapClient = HubNewEapClient(p->Cedar, p->Eap_Identity.HubName, client_ip_tmp, p->Eap_Identity.UserName, "L3:PPP", true, 
+													&response, pp->Lcp->Id);
 
-					if (p->EapClient == NULL)
+					if (p->EapClient == NULL || response == NULL)
 					{
 						PPP_PACKET *pack = ZeroMalloc(sizeof(PPP_PACKET));
 						pack->IsControl = true;
 						pack->Protocol = PPP_PROTOCOL_EAP;
 						PPPSetStatus(p, PPP_STATUS_AUTH_FAIL);
 						pack->Lcp = NewPPPLCP(PPP_EAP_CODE_FAILURE, p->Eap_PacketId);
+						Debug("Failed to connect to a RADIUS server\n");
 
 						if (PPPSendPacketAndFree(p, pack) == false)
 						{
 							PPPSetStatus(p, PPP_STATUS_FAIL);
 							WHERE;
 							return false;
 						}
-						break;
 					}
+					else
+					{
+						// Send first response to client
+						if (PPPSendAndRetransmitRequest(p, PPP_PROTOCOL_EAP, response) == false)
+						{
+							PPPSetStatus(p, PPP_STATUS_FAIL);
+							WHERE;
+							return false;
+						}
+					}	
+
+					break;

```

Comment:
This block of code is better placed at line 245, setting PPP_STATUS_BEFORE_AUTH here only.
Now, is there a way to delay the first Radius response to a later date?